commit ea0c9925c114b6d035b69ccfd165641f1a0eb813
Author: Nir Dotan <nird@mellanox.com>
Date:   Fri Aug 3 15:57:44 2018 +0300

    mlxsw: core_acl_flex_actions: Remove redundant mirror resource destruction
    
    [ Upstream commit caebd1b389708bf3d0465be829480fc706a68720 ]
    
    In previous patch mlxsw_afa_resource_del() was added to avoid a duplicate
    resource detruction scenario.
    For mirror actions, such duplicate destruction leads to a crash as in:
    
     # tc qdisc add dev swp49 ingress
     # tc filter add dev swp49 parent ffff: \
       protocol ip chain 100 pref 10 \
       flower skip_sw dst_ip 192.168.101.1 action drop
     # tc filter add dev swp49 parent ffff: \
       protocol ip pref 10 \
       flower skip_sw dst_ip 192.168.101.1 action goto chain 100 \
       action mirred egress mirror dev swp4
    
    Therefore add a call to mlxsw_afa_resource_del() in
    mlxsw_afa_mirror_destroy() in order to clear that resource
    from rule's resources.
    
    Fixes: d0d13c1858a1 ("mlxsw: spectrum_acl: Add support for mirror action")
    Signed-off-by: Nir Dotan <nird@mellanox.com>
    Reviewed-by: Jiri Pirko <jiri@mellanox.com>
    Signed-off-by: Ido Schimmel <idosch@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c b/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c
index a54f23f00a5f..f6f6a568d66a 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/core_acl_flex_actions.c
@@ -862,6 +862,7 @@ static void
 mlxsw_afa_mirror_destroy(struct mlxsw_afa_block *block,
 			 struct mlxsw_afa_mirror *mirror)
 {
+	mlxsw_afa_resource_del(&mirror->resource);
 	block->afa->ops->mirror_del(block->afa->ops_priv,
 				    mirror->local_in_port,
 				    mirror->span_id,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1012_ea0c9925c114b6d035b69ccfd165641f1a0eb813.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1012_ea0c9925c114b6d035b69ccfd165641f1a0eb813.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
